src/System.ComponentModel.TypeConverter/src/System/Timers/Timer.cs

```diff
@@ -0,0 +1,350 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading;
+using System.ComponentModel;
+using System.ComponentModel.Design;
+using System.Security.Permissions;
+
+namespace System.Timers
+{
+    /// <summary>
+    ///    <para>Handles recurring events in an application.</para>
+    /// </summary>
+    [DefaultProperty("Interval"), DefaultEvent("Elapsed"), HostProtection(Synchronization = true, ExternalThreading = true)]
+    public partial class Timer : Component, ISupportInitialize
+    {
+        private double _interval;
+        private bool _enabled;
+        private bool _initializing;
+        private bool _delayedEnable;
+        private ElapsedEventHandler _onIntervalElapsed;
+        private bool _autoReset;
+        private ISynchronizeInvoke _synchronizingObject;
+        private bool _disposed;
+        private Threading.Timer _timer;
+        private TimerCallback _callback;
+        private object _cookie;
+
+        /// <summary>
+        /// <para>Initializes a new instance of the <see cref='System.Timers.Timer'/> class, with the properties
+        ///    set to initial values.</para>
+        /// </summary>
+        public Timer()
+        : base()
+        {
+            _interval = 100;
+            _enabled = false;
+            _autoReset = true;
+            _initializing = false;
+            _delayedEnable = false;
+            _callback = new TimerCallback(MyTimerCallback);
+        }
+
+        /// <summary>
+        ///    <para>
+        ///       Initializes a new instance of the <see cref='System.Timers.Timer'/> class, setting the <see cref='System.Timers.Timer.Interval'/> property to the specified period.
+        ///    </para>
+        /// </summary>
+        public Timer(double interval)
+        : this()
+        {
+            if (interval <= 0)
+                throw new ArgumentException(SR.Format(SR.InvalidParameter, "interval", interval));
+
+            double roundedInterval = Math.Ceiling(interval);
+            if (roundedInterval > int.MaxValue || roundedInterval <= 0)
+            {
+                throw new ArgumentException(SR.Format(SR.InvalidParameter, "interval", interval));
+            }
+
+            _interval = (int)roundedInterval;
+        }
+        
+        /// <summary>
+        /// <para>Gets or sets a value indicating whether the Timer raises the Tick event each time the specified
+        /// Interval has elapsed,
+        ///    when Enabled is set to true.</para>
+        /// </summary>
+        [TimersDescription(nameof(SR.TimerAutoReset), null), DefaultValue(true)]
+        public bool AutoReset
+        {
+            get
+            {
+                return _autoReset;
+            }
+
+            set
+            {
+                if (DesignMode)
```

Comment:
If we've hardcoded DesignMode to return false and it's not virtual, do we need any of the code protected by ifs for DesignMode?